1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js

```diff
@@ -11,13 +11,8 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
 ------------------------------------------------------------------------------*/
 // ! Function to be tested
 function doubleEvenNumbers(numbers) {
-  // TODO rewrite the function body using `map` and `filter`.
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
+  let newNumbers = [];
```

Comment:
I don't think this instantiation is necessary because you're not appending to the array, you're over writing the value thus using unnecessary memory. Fix is to simply do
```suggestion
  let newNumbers = numbers.filter(number=>number%2===0).map(number=>number*2);
```

Reply:
i deleted assignment and implemented code block.